### ISSUE TITLE
Tool tip shadow not transparent when over heavyweight component

### ISSUE THREAD

**Issue by rkeen-siemens** [JFormDesigner/FlatLaf#626]
When a tooltip is displayed over a heavyweight component the shadow is on a white background rather than transparent.
<img width="183" alt="Within Window Bounds" src="https://user-images.githubusercontent.com/46196614/206799389-93efc884-71a2-4b9a-810b-ebb6f7a8b00a.png">

If the tooltip is partly outside the bounds of the main window, it is rendered as expected.
<img width="183" alt="Outside Window Bounds" src="https://user-images.githubusercontent.com/46196614/206799388-9efe1027-836b-4da6-847d-6cb178df7717.png">

I suspect this is due to the fact that the `PopupFactory` is not forcing a heavyweight component when the popup will be displayed over a heavyweight component. This is probably a general issue in Swing but it is not seen there since the default tool tip popup doesn't have a drop shadow. Since FlatLaf returns a `DropShadowPopup` it becomes an issue.

I understand I can disable the drop shadow (via "Popup.dropShadowPainted") or force heavy weight popups (via "Popup.forceHeavyWeight"), but it would be nice if this worked out of the box. Perhaps force a heavy weight popup if there are heavyweight components at the extents of the tool tip?

Here's the source to reproduce. This seems to be an issue on Windows, but not Linux (didn't try Mac).

```java
import java.awt.BorderLayout;
import java.awt.Button;
import java.awt.Color;
import java.awt.Dimension;

import javax.swing.JButton;
import javax.swing.SwingUtilities;
import javax.swing.WindowConstants;

import com.formdev.flatlaf.FlatLightLaf;

public class Form extends javax.swing.JFrame {
	public Form() {
		setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
		setTitle("Tool Tip Over Heavyweight");
		getContentPane().setLayout(new BorderLayout(4, 4));

		JButton lightweight = new JButton("Lightweight");
		lightweight.setToolTipText("Lightweight");
		getContentPane().add(lightweight, BorderLayout.PAGE_START);

		Button heavyweight = new Button("Heavyweight");
		heavyweight.setBackground(Color.BLACK);
		heavyweight.setForeground(Color.YELLOW);
		getContentPane().add(heavyweight, BorderLayout.CENTER);
		setMinimumSize(new Dimension(150, 200));
		setLocationRelativeTo(null);
		pack();
	}

	public static void main(String args[]) {
		FlatLightLaf.setup();
		SwingUtilities.invokeLater(() -> new Form().setVisible(true));
	}
}
```

**Comment by DevCharly** [comment]
At the moment I would recommend to force heavyweight popups for your application:

~~~java
UIManager.put( "Popup.forceHeavyWeight", true );
~~~

Lightweight popups are only used on Windows and only if the popup fits into the window.
On macOS and on Linux, always heavyweight popups are used because they provide operating system border and drop shadow.

> ...but it would be nice if this worked out of the box. Perhaps force a heavy weight popup if there are heavyweight components at the extents of the tool tip?

I'll try this...

BTW working a way to use popups with native borders and drop shadow on Windows 11:

![image](https://user-images.githubusercontent.com/5604048/207683533-05ebb768-f6ce-4f8e-9148-a5be61f8692d.png) ![image](https://user-images.githubusercontent.com/5604048/207683582-84514cb9-481a-418d-9b46-f17bfb2b9bc8.png) ![image](https://user-images.githubusercontent.com/5604048/207686187-235bbe62-1b14-4b3a-8704-d3d89b583896.png)

Then also heavyweight popup will be used on Windows 11.
Only Window 10 will use lightweight popups.

**Comment by DevCharly** [comment]
fixed in latest `3.3-SNAPSHOT`: https://github.com/JFormDesigner/FlatLaf#snapshots

**Comment by DevCharly** [comment]
fixed in FlatLaf 3.2.1
